Config Kotlin, Implement SignUp features

thachnguyendino opened the pull request:
- Changes:
   + Converted all java class to kotlin class
   + Implement facebook login (90% done)
   + Implement sign up login (40% done)
   + Added dagger 2
   + Added RxBinding.

thachnguyendino committed:
Config Kotlin, Implement SignUp features
- Changes:
   + Converted all java class to kotlin class
   + Implement facebook login (90% done)
   + Implement sign up login (40% done)
   + Added dagger 2
   + Added RxBinding.

thachnguyendino committed:
Implement sign in feature.
Changes:
- Created BaseFragment.kt to manage disposable from Rx
- Added Dagger to handle component and module. (di package)
- Create Authentication API for (signIn, signUp, signOut)
- Create SignInModel to store data respone for sign in feature.
- Upgrade gradle to version 3.5 because dagger 2 required.

thachnguyendino committed:
- Removed unsage variables.
- Correct function name.

thachnguyendino committed:
Implement sign up.
- Handle errors when user input data in sign up screen.

thachnguyendino committed:
Changes: Refator model class.

ndduy commented on SportBook/app/build.gradle:
```diff
@@ -11,37 +12,74 @@ android {
         versionName "1.0"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
     }
+    signingConfigs {
```
Please create different flavors for different end point as well.

We are going to have Staging and Production Flavors

ndduy commented on SportBook/app/src/main/java/com/dinosys/sportbook/application/SportbookApp.kt:
```diff
@@ -0,0 +1,22 @@
+class SportbookApp: Application() {
+
+    companion object {
+        lateinit var authComponent: AuthenticationComponent
```
👍

ndduy commented on SportBook/app/src/main/java/com/dinosys/sportbook/features/signin/SignInFragment.kt:
```diff
@@ -0,0 +1,111 @@
+                            }
+                }
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ response -> onSignInDataResponse(response = response) })
```
What is the convention? Where should we use RxJava? Where should we use dagger

ndduy commented on SportBook/app/src/main/java/com/dinosys/sportbook/features/signin/SignInViewModel.kt:
```diff
@@ -0,0 +1,38 @@
+        }
+        return authApi.signIn(userName, password)
+    }
+}
```
Why do we use Singleton for the Sign In Model

ndduy commented on SportBook/app/src/main/java/com/dinosys/sportbook/utils/ConextUtil.kt:
```diff
@@ -0,0 +1,26 @@
+import android.support.v4.app.Fragment
+import android.view.View
+
+object ConextUtil {
```
Typo

thachnguyendino committed:
Changes:
- Created different flavors for different end point (Staging/Production)
- Removed @Singleton anotation in SignInViewModel class.
- Correct ContextUtil class name.

thachnguyendino committed:
- Changes (add, remove) function in  FragmentManager class to extension feature
 to easy navigator beetween fragment and activity
